ambersim/utils/io_utils.py

```diff
@@ -0,0 +1,230 @@
+
+    See: https://stackoverflow.com/a/66552066
+    """
+    if f.is_file():
+        f.unlink()
+    else:
+        for child in f.iterdir():
+            _rmtree(child)
+        f.rmdir()
+
+
+# ############# #
+# MODEL LOADING #
+# ############# #
+
+
+def _modify_robot_float_base(filepath: Union[str, Path]) -> mj.MjModel:
+    """Modifies a robot to have a floating base if it doesn't already."""
+    # loading current robot
+    assert str(filepath).split(".")[-1] == "xml"
+    robot = mjcf.from_file(filepath, model_dir="/".join(filepath.split("/")[:-1]))
+
+    # only add free joint if the first body after worldbody has no freejoints
+    assert robot.worldbody is not None
+    if robot.worldbody.body[0].freejoint is None:
+        # creates a new robot object and attaches the current robot to it freely
+        arena = mjcf.RootElement(model=robot.model)
+        attachment_frame = arena.attach(robot)
+        attachment_frame.add("freejoint", name="freejoint")
+        robot = arena
+
+        # ensuring that the base link has inertial attributes
+        if robot.worldbody.body[0].inertial is None:
+            robot.worldbody.body[0].add("inertial", mass=1e-6, diaginertia=1e-6 * np.ones(3), pos=np.zeros(3))
+
+    # extracts the mujoco model from the dm_mujoco Physics object
+    physics = mjcf.Physics.from_mjcf_model(robot)
+    model = physics.model.ptr
+    return model
+
+
+def load_mj_model_from_file(filepath: Union[str, Path], force_float: bool = False) -> mj.MjModel:
+    """Loads a mujoco model from a filepath.
+
+    Args:
+        filepath: A path to a URDF or MJCF file. This can be global, local, or with respect to the repository root.
+        force_float: Whether to forcibly float the base of the robot.
+
+    Returns:
+        mj_model: A mujoco model.
+    """
+    filepath = _check_filepath(filepath)
+
+    # loading the model and data. check whether freejoint is added forcibly
+    if force_float:
+        # check that the file is an XML. if not, save as xml temporarily
+        if str(filepath).split(".")[-1] != "xml":
+            output_name = "/".join(str(filepath).split("/")[:-1]) + "/_temp_xml_model.xml"
+            save_model_xml(filepath, output_name=output_name)
+            mj_model = _modify_robot_float_base(output_name)
+            Path.unlink(output_name)
+        else:
+            mj_model = _modify_robot_float_base(filepath)
+    else:
+        mj_model = mj.MjModel.from_xml_path(filepath)
+    return mj_model
+
+
+def mj_to_mjx_model(mj_model: mj.MjModel) -> Tuple[mjx.Model, mjx.Data]:
+    """Converts a mujoco model to an mjx (model, data) pair."""
+    try:
+        mjx_model = mjx.device_put(mj_model)
+        mjx_data = mjx.make_data(mj_model)
+        return mjx_model, mjx_data
+    except NotImplementedError as e:
+        print()
+        print("There are some URDF convex primitives that aren't compatible with mjx's convex collision checking.")
```

Comment:
TODO(ahl): As per offline conversation with @pculbertson, this extended print statement should be expanded to include other possible causes for throwing the error, including friction cone stuff. 

Reference this feature parity table: https://mujoco.readthedocs.io/en/latest/mjx.html#feature-parity

Reply:
Done.